src/runtime_src/core/pcie/driver/linux/xocl/subdev/mailbox.c

```diff
@@ -1583,8 +1583,10 @@ static bool is_tx_chan_ready(struct mailbox_channel *ch)
 	bool sw_ready, hw_ready;
 	u32 st;
 
+	mutex_lock(&ch->sw_chan_mutex);
 	sw_ready = (ch->sw_chan_msg_id == 0);
-	if (MB_SW_ONLY(mbx))
+	mutex_unlock(&ch->sw_chan_mutex);
+	if ((ch->mbc_cur_msg && ch->mbc_cur_msg->mbm_chan_sw) || MB_SW_ONLY(mbx))	
```

Comment:
this change will violate the rule we set up (see comment at line 1596). do you really need this change?